packages/kbn-optimizer/src/worker/webpack.config.ts

```diff
@@ -135,6 +135,14 @@ export function getWebpackConfig(bundle: Bundle, worker: WorkerConfig) {
                         return null;
                       }
 
+                      if (uri.startsWith('ui/assets')) {
+                        return Path.resolve(
+                          worker.repoRoot,
+                          'src/core/server/core_app/',
+                          uri.replace('ui/', '')
+                        );
+                      }
+
                       // manually force ui/* urls in legacy styles to resolve to ui/legacy/public
                       if (uri.startsWith('ui/') && parseDirPath(base).dirs.includes('legacy')) {
```

Comment:
Let's move the `legacy` directory check higher up so that it applies to both of these. We don't want to accidentally start depending on request rewriting in the NP.